src/controllers/dynakube/connectioninfo/communication_hosts.go

```diff
@@ -16,3 +25,63 @@ func GetOneAgentCommunicationHosts(dynakube *dynatracev1beta1.DynaKube) []dtclie
 	}
 	return communicationHosts
 }
+
+func GetActiveGateEndpointsAsCommunicationHosts(dynakube *dynatracev1beta1.DynaKube) []dtclient.CommunicationHost {
+	activegateEndpointsString := dynakube.Status.ActiveGate.ConnectionInfoStatus.Endpoints
+	if activegateEndpointsString == "" {
+		return []dtclient.CommunicationHost{}
+	}
+	return parseCommunicationHostFromActiveGateEndpoints(activegateEndpointsString)
+}
+
+func parseCommunicationHostFromActiveGateEndpoints(activegateEndpointsString string) []dtclient.CommunicationHost {
+	endpointStrings := strings.Split(activegateEndpointsString, ",")
+	endpointCommunicationHosts := make(map[dtclient.CommunicationHost]bool, len(endpointStrings))
+	for _, endpointString := range endpointStrings {
+		if endpoint, err := parseEndpointToCommunicationHost(endpointString); err == nil {
+			endpointCommunicationHosts[endpoint] = true
+		}
+	}
+
+	comHosts := make([]dtclient.CommunicationHost, 0, len(endpointCommunicationHosts))
+	for ch := range endpointCommunicationHosts {
+		comHosts = append(comHosts, ch)
+	}
+
+	return comHosts
+}
+
+func parseEndpointToCommunicationHost(endpointString string) (dtclient.CommunicationHost, error) {
+	if endpointString == "" {
+		return dtclient.CommunicationHost{}, errors.New("empty endpoint string not allowed")
+	}
+	parsedEndpoint, err := url.Parse(endpointString)
+
+	if err != nil {
+		return dtclient.CommunicationHost{}, err
+	}
+	host := parsedEndpoint.Host
+	parts := strings.Split(host, ":")
+
+	port, err := parsePort(parts)
```

Comment:
I think you can reuse standart library: `net.SplitHostPort` https://pkg.go.dev/net#SplitHostPort and remove parsePort function.

Reply:
that looks rather nice